src/extra/gpkg.jl

```diff
@@ -187,3 +187,365 @@ function skipgpkgheader!(buff)
   # Float64[minx, maxx, miny, maxy, minz, maxz] => skip 48 bytes
   E > 0 && skip(buff, 8 * 2 * (E + 1))
 end
+
+# According to https://www.geopackage.org/spec/#r2
+# a GeoPackage should contain "GPKG" in ASCII in
+# "application_id" field of SQLite db header
+const GPKG_APPLICATION_ID = 0x47504B47
+const GPKG_1_4_VERSION = 10400
+
+# If the geometry type_name value is "GEOMETRY" then the feature table geometry column
+# MAY contain geometries of any allowed geometry type
+function SQLite.sqlitetype_(::Type{Vector{UInt8}})
+  "GEOMETRY"
+end
+
+function gpkgwrite(fname, geotable;)
+  db = SQLite.DB(fname)
+
+  # https://sqlite.org/pragma.html#pragma_synchronous
+  # Commits can be orders of magnitude faster with
+  # Setting PRAGMA synchronous=OFF but,
+  # can cause the database to go corrupt
+  # if there is an operating system crash or power failure.
+  DBInterface.execute(db, "PRAGMA synchronous=0")
+
+  table = values(geotable)
+  domain = GeoTables.domain(geotable)
+  crs = GeoTables.crs(domain)
+  geom = collect(domain)
+
+  # According to https://www.geopackage.org/spec/#r2
+  # A GeoPackage SHALL contain a value of 0x47504B47 ("GPKG" in ASCII)
+  # in the "application_id" field and an appropriate value in "user_version" field
+  # of the SQLite database header to indicate that it is a GeoPackage
+  DBInterface.execute(db, "PRAGMA application_id = $GPKG_APPLICATION_ID ")
+  DBInterface.execute(db, "PRAGMA user_version = $GPKG_1_4_VERSION ")
+
+  creategpkgtables(db, table, domain, crs, geom)
+
+  # https://sqlite.org/pragma.html#pragma_optimize
+  # Applications with short-lived database connections should run "PRAGMA optimize;"
+  # just once, prior to closing each database connection.
+  DBInterface.execute(db, "PRAGMA optimize;")
+
+  DBInterface.close!(db)
+end
+
+function creategpkgtables(db, table, domain, crs, geoms)
+  if crs <: Cartesian
+    org = ""
+    srsid = -1
+  else
+    org = "EPSG"
+    srsid = first(CoordRefSystems.code(crs).parameters)
```

Comment:
Codes don't have a `.parameters` field. That is why I am assuming you are using some AI to write things down. Otherwise, you are creating APIs out of the air.

Reply:
Added a `CoordRefSystems.integer` function to the CRSCode API: https://github.com/JuliaEarth/CoordRefSystems.jl/pull/333